### PR TITLE
Search support in the blocks editor

### LINKED ISSUE #1000: Add search functionality to the blocks editor
Add search bar in the blocks editor toolbox. 
Typing in the search bar filters the toolbox for blocks that match the search term.

### PR DESCRIPTION
**Pull request by microsoftsam**
pxtsearch

- Search bar on the top of the Blockly Toolbox 
- Search debounce is 200 ms 
- Search icon turns loading on when search is running 
- Search runs in the web worker 
- Search results show in a "Search" category that only shows up when there is a search term in the input
- Search results are limited to 7 blocks
- Search input style can be customized in target using: '.blocklySearchInputField'

Fixes #1000

### CODE REVIEW

File: pxtlib/emitter/service.ts [hunk 1]
```diff
@@ -717,9 +728,38 @@ namespace ts.pxtc.service {
...
+            if (!fuse) fuse = new Fuse(blockInfo.blocks, fuseOptions);
+
+            const sorter = (fn: pxtc.SymbolInfo): number => {
+                // sort by namespace weight
+                let nsn = blockInfo.apis.byQName[fn.namespace.split('.')[0]];
+                return nsn && nsn.attributes ? nsn.attributes.weight || 50 : 50;
+            }
+            const fns = fuse.search(search)
+                .slice(0, SEARCH_RESULT_COUNT)
+                .sort((l, r) => - sorter(l) + sorter(r));
```
Comment (pelikhan) [inline pxtlib/emitter/service.ts]: Do we still need to sort? This should be covered by the fuse weights.

File: theme/pxt.less [hunk 2]
```diff
@@ -249,6 +249,14 @@ div.simframe > iframe {
     fill: none !important;
 }
 
+.blocklyFlyoutLabelText {
+    fill: white !important;
```
Comment (pelikhan) [inline theme/pxt.less]: Theming? @white?

File: webapp/src/blocks.tsx [hunk 3]
```diff
@@ -514,3 +522,12 @@ export class Editor extends srceditor.Editor {
         blocks.filter(b => b.isShadow_).forEach(b => b.dispose(false));
     }
 }
+
+// block-search:<searchFor>
+data.mountVirtualApi("block-search", {
```
Comment (pelikhan) [inline webapp/src/blocks.tsx]: Not clear we still need to get through this

File: theme/pxt.less (hunk 2 above)
Comment (microsoftsam) [inline theme/pxt.less]: theming can override this. The default fill is black for blockly, and it's more likely white would have better contrast here, hence why it's in pxt.less.

**Comment by pelikhan** [comment]
LGTM

**Comment by Jaqster** [comment]
Looks awesome!

**Pull request merged**